lib/loaders/api/loader_interface.js

```diff
@@ -0,0 +1,32 @@
+import { toKey } from "lib/helpers"
+
+const encodeStaticPath = (path, globalParams, params) => {
+  return toKey(path, Object.assign({}, globalParams, params))
+}
+
+const encodeDynamicPath = (pathGenerator, globalParams, id, params) => {
+  return encodeStaticPath(pathGenerator(id), globalParams, params)
+}
+
+/**
+ * This implements the common interface for producing data loaders for each api/path, regardless of authentication.
+ *
+ * If `path` is a string, then the loader will accept only query params as its parameters. For example:
+ *
+ *    const artworksLoader = gravityLoader("artworks")
+ *    artworksLoader({ page: 42 })
+ *
+ * If `path` is a function, then the loader accepts an ID _and_ query params as its parameters. For example:
+ *
+ *    const artistArtworksLoader = gravityLoader(id => `artist/${id}/artworks`)
```

Comment:
Your comment reminded me that we had talked about this API before https://github.com/artsy/metaphysics/pull/572#issuecomment-287129665

Would you prefer the following?

```js
const saleArtworkLoader = gravityLoader("sale/:sale_id/sale_artwork/:id")
saleArtworkLoader({ id: artwork.id, sale_id: sale.id })
```

That would standardise the API to always just accept 1 dictionary parameter, my only concern is what to do with the params that we interpolated into the path, should we take this out of the dictionary that we post to the API or just leave them in and assume the API ignores them?